base/shell/explorer/trayprop.cpp

```diff
@@ -250,16 +253,58 @@ class CStartMenuSettingsPage : public CPropertyPageImpl<CStartMenuSettingsPage>
 
     LRESULT OnInitDialog(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
     {
+        HKEY hKey;
+        DWORD dSz;
+        BYTE regData[12];
+        DWORD dType;
         // fix me: start menu style (classic/modern) should be read somewhere from the registry.
-        CheckDlgButton(IDC_TASKBARPROP_STARTMENUCLASSIC, BST_CHECKED); // HACK: This has to be read from registry!!!!!!!
+       if( RegOpenKeyEx(HKEY_CURRENT_USER, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\Explorer",
+                          0, KEY_QUERY_VALUE|KEY_SET_VALUE, &hKey) == ERROR_SUCCESS)
+        {
+            RegQueryValueEx(hKey, L"noSampleStartMenu", NULL, &dType, regData,&dSz );
+            RegCloseKey(hKey);
+        }
+        if(*((DWORD*)&regData[0])) CheckDlgButton(IDC_TASKBARPROP_STARTMENU, BST_CHECKED);
+        else CheckDlgButton(IDC_TASKBARPROP_STARTMENUCLASSIC, BST_CHECKED);
         UpdateDialog();
     
         return TRUE;
     }
 
+    LRESULT OnStartMenuSelection(WORD wNotifyCode, WORD wID, HWND hWndCtl, BOOL &bHandled)
+    {
+        if(wNotifyCode == BN_CLICKED)
+        {
+                 HWND hwndStartBitmap = GetDlgItem(IDC_TASKBARPROP_STARTMENU_BITMAP);
+                 HWND hwndCustomizeClassic = GetDlgItem(IDC_TASKBARPROP_STARTMENUCLASSICCUST);
+                 HWND hwndCustomizeModern = GetDlgItem(IDC_TASKBARPROP_STARTMENUCUST);
+                 if(wID == IDC_TASKBARPROP_STARTMENU)
+                 {
+                      CheckDlgButton(IDC_TASKBARPROP_STARTMENU, BST_CHECKED);
+                      CheckDlgButton(IDC_TASKBARPROP_STARTMENUCLASSIC, BST_UNCHECKED);
+                    ::EnableWindow(hwndCustomizeModern, TRUE);
```

Comment:
Indentation problem.

Reply:
When I do not know, I ask.
What is identation?